driver-core/src/main/com/mongodb/ClientSessionOptions.java

```diff
@@ -224,8 +224,9 @@ public Builder defaultTransactionOptions(final TransactionOptions defaultTransac
          * @see #getDefaultTimeout
          */
         public Builder defaultTimeout(final long defaultTimeout, final TimeUnit timeUnit) {
-            isTrueArgument("timeoutMS must be >= 0", defaultTimeout >= 0);
-            this.defaultTimeoutMS = MILLISECONDS.convert(defaultTimeout, timeUnit);
+            this.defaultTimeoutMS = isTrueArgument("timeoutMS must be >= 0",
```

Comment:
To enhance clarity, we could update the validation message to be more descriptive, specifying the parameter name and differentiating between the two scenarios where validation fails. Currently, we use a single message for both cases.

I propose two distinct messages:
For a too small defaultTimeout: `defaultTimeout is too small. It was rounded to 0 milliseconds, resulting in an unintended infinite timeout.`
For a negative value after conversion: `defaultTimeout, after conversion to milliseconds, must be >= 0.`

This suggestion applies to all timeout/withTimeout methods.
What do you think?

Reply:
SGTM!